monitoring-dashboard/components/org.wso2.micro.integrator.dashboard.web/web-app/src/commons/sideDrawers/DataSourcesSideDrawer.js

```diff
@@ -0,0 +1,112 @@
+ * specific language governing permissions and limitations
+ * under the License.
+ *
+ *
+ */
+
+import React from 'react';
+import { makeStyles } from '@material-ui/core/styles';
+import Paper from '@material-ui/core/Paper';
+import Box from '@material-ui/core/Box';
+import Grid from '@material-ui/core/Grid';
+import { Table, TableCell, TableRow } from '@material-ui/core';
+import HeadingSection from './commons/HeadingSection'
+import SourceViewSection from './commons/SourceViewSection'
+import Typography from '@material-ui/core/Typography';
+
+export default function DataSourcesSideDrawer(props) {
+    const { nodeData } = props;
+    const nodeId = nodeData.nodeId;
+    const artifactName = nodeData.details.name;
+    const classes = useStyles();
+
+    return (
+        <div className={classes.root}>
+            <Grid container spacing={3}>
+                <Grid item xs={12}>
+                    <HeadingSection name={artifactName} nodeId={nodeId} />
+                    <SourceViewSection
+                        artifactType="data-sources" artifactName={artifactName} nodeId={nodeId}
+                        designContent={<>
+                            <Paper className={classes.paper} elevation={0} square>
+                                <DataSourceDetailTable nodeData={nodeData.details} />
+                            </Paper>
+                            <Box pl={4}>
+                                <Typography variant="h6" color="inherit" noWrap>
+                                    Parameters
+                                </Typography>
+                                <Box pr={2}>
+                                    <ParametersSection parameters={nodeData.details.configurationParameters} />
+                                </Box>
+                            </Box>
+                        </>}
+                    />
+                </Grid>
+            </Grid>
+        </div>
+    );
+}
+
+function DataSourceDetailTable(props) {
+    const nodeData = props.nodeData;
+    return <Table>
+        <TableRow>
+            <TableCell>Type</TableCell>
+            <TableCell>{nodeData.type}</TableCell>
+        </TableRow>
+        <TableRow>
+            <TableCell>Description</TableCell>
+            <TableCell>{nodeData.description}</TableCell>
+        </TableRow>
+        <TableRow>
+            <TableCell>Driver</TableCell>
+            <TableCell>{nodeData.driverClass}</TableCell>
+        </TableRow>
+        <TableRow>
+            <TableCell>URL</TableCell>
+            <TableCell>{nodeData.url}</TableCell>
+        </TableRow>
+        <TableRow>
+            <TableCell>Username</TableCell>
+            <TableCell>{nodeData.userName}</TableCell>
+        </TableRow>
+    </Table>
+}
+
+function ParametersSection(props) {
+    const parameters = props.parameters;
+    return <Table size="small" style={{width: '100%'}}>
+        {Object.entries(parameters).map(([key, value]) => 
+            <TableRow>
```

Comment:
Shall we try to define a key for this loop [1]? We might be getting a small and less frequently updating list now, but if we have defined a key, if this list grows in the future, there won't be any performance degradation. 

[1] - https://reactjs.org/docs/lists-and-keys.html#keys

Reply:
Fixed with 795cc4d